sequencer/src/state.rs

```diff
@@ -298,10 +298,20 @@ impl HotShotState<SeqTypes> for ValidatedState {
         let mut validated_state = self.clone();
 
         // Fetch the new L1 deposits between parent and current finalized L1 block.
-        let l1_deposits = fetch_fee_receipts(
-            parent_leaf.get_block_header().l1_finalized,
-            proposed_header.l1_finalized,
-        );
+
+        let l1_deposits = instance
+            .l1_client
+            .get_finalized_deposits(
+                parent_leaf
+                    .get_block_header()
+                    .l1_finalized
+                    .map(|block_info| block_info.number),
+                proposed_header
+                    .l1_finalized
+                    .map(|block_info| block_info.number)
+                    .unwrap_or(0),
```

Comment:
Same comment here - I don't think it ever makes sense to resort to a block height of 0, we certainly wont have any deposits there in production.

Reply:
fixed here: 5c8cab9d2d9b434f3b9ba0a68e540d582a719e83